File: server/internal/model/exercise.go
package model

import (
	"time"

	"github.com/google/uuid"

	"wsTrack/server/internal/enum"
)

type Exercise struct {
	ID               uuid.UUID  `gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	UserID           *uuid.UUID `gorm:"type:uuid;index"`
	User             *User      `gorm:"foreignKey:UserID"`
	Name             string     `gorm:"not null"`
	NameEn           *string
	Category         enum.ExerciseCategory `gorm:"type:varchar(32);not null"`
	PrimaryMuscles   StringArray           `gorm:"type:text[];not null"`
	SecondaryMuscles StringArray           `gorm:"type:text[]"`
	Equipment        enum.Equipment        `gorm:"type:varchar(32);not null"`
	TrackingType     enum.TrackingType     `gorm:"type:varchar(32);not null"`
	UnitPreference   *enum.WeightUnit      `gorm:"type:varchar(8)"`
	IsCustom         bool                  `gorm:"default:false"`
	IsArchived       bool                  `gorm:"default:false"`
	Notes            *string
	SortOrder        int `gorm:"default:0"`
	CreatedAt        time.Time
	UpdatedAt        time.Time
}

File: mobile/src/models/Exercise.ts
import { Model } from "@nozbe/watermelondb";
import type { Query } from "@nozbe/watermelondb";
import type { Associations } from "@nozbe/watermelondb/Model";
import { children, field, json } from "@nozbe/watermelondb/decorators";

import type {
  Equipment,
  ExerciseCategory,
  MuscleGroup,
  TrackingType,
  WeightUnit,
} from "../constants/enums";
import { sanitizeMuscleGroups } from "./sanitizers";
import type { PersonalRecord } from "./PersonalRecord";
import type { PlanExercise } from "./PlanExercise";
import type { TemplateExercise } from "./TemplateExercise";
import type { WorkoutExercise } from "./WorkoutExercise";

export class Exercise extends Model {
  static table = "exercises";

  static associations: Associations = {
    plan_exercises: { type: "has_many", foreignKey: "exercise_id" },
    personal_records: { type: "has_many", foreignKey: "exercise_id" },
    template_exercises: { type: "has_many", foreignKey: "exercise_id" },
    workout_exercises: { type: "has_many", foreignKey: "exercise_id" },
  };

  @field("name") name!: string;
  @field("name_en") nameEn!: string | null;
  @field("category") category!: ExerciseCategory;
  @json("primary_muscles", sanitizeMuscleGroups) primaryMuscles!: MuscleGroup[];
  @json("secondary_muscles", sanitizeMuscleGroups) secondaryMuscles!: MuscleGroup[];
  @field("equipment") equipment!: Equipment;
  @field("tracking_type") trackingType!: TrackingType;
  @field("unit_preference") unitPreference!: WeightUnit | null;
  @field("is_custom") isCustom!: boolean;
  @field("is_archived") isArchived!: boolean;
  @field("notes") notes!: string | null;
  @field("sort_order") sortOrder!: number;
  @field("created_at") createdAt!: number;
  @field("updated_at") updatedAt!: number;

  @children("plan_exercises") planExercises!: Query<PlanExercise>;
  @children("personal_records") personalRecords!: Query<PersonalRecord>;
  @children("template_exercises") templateExercises!: Query<TemplateExercise>;
  @children("workout_exercises") workoutExercises!: Query<WorkoutExercise>;
}
